packages/react-native-bridge/common/gutenberg-web-single-block/editor-behavior-overrides.js

```diff
@@ -1,59 +1,167 @@
+			if (
+				! isContextMenuVisible ||
+				! selected ||
+				! selected.toString()
+			) {
+				return;
 			}
-		}
 
-		// Hide text selection context menu when the click
-		// is triggered by a dropdown toggle.
-		//
-		// NOTE: The event propagation is prevented because
-		// it will be dispatched after the context menu
-		// is hidden.
-		if ( currentToggle ) {
-			event.stopPropagation();
-			hideContextMenuListeners.push( () => currentToggle.click() );
-			window.wpwebkit.hideTextSelectionContextMenu();
+			// Check if the event is triggered by a dropdown
+			// toggle button.
+			const dropdownToggles = document.querySelectorAll(
+				'.components-dropdown-menu > button'
+			);
+			let currentToggle;
+			for ( const node of dropdownToggles.values() ) {
+				if ( node.contains( event.target ) ) {
+					currentToggle = node;
+					break;
+				}
+			}
+
+			// Hide text selection context menu when the click
+			// is triggered by a dropdown toggle.
+			//
+			// NOTE: The event propagation is prevented because
+			// it will be dispatched after the context menu
+			// is hidden.
+			if ( currentToggle ) {
+				event.stopPropagation();
+				hideContextMenuListeners.push( () => currentToggle.click() );
+				window.wpwebkit.hideTextSelectionContextMenu();
+			}
+		},
+		true
+	);
+}
+
+if ( isAndroid() ) {
+	manageTextSelectonContextMenu();
+}
+
+const editor = document.querySelector( '#editor' );
+
+function _toggleBlockSelectedClass( isBlockSelected ) {
+	if ( isBlockSelected ) {
+		editor.classList.add( 'is-block-selected' );
+	} else {
+		editor.classList.remove( 'is-block-selected' );
+	}
+}
+
+/** @typedef {import('@wordpress/data').WPDataRegistry} WPDataRegistry */
+
+/**
+ * Toggle the `is-block-selected` class on the editor container when a block is
+ * selected. This is used to hide the sidebar toggle button when a block is not
+ * selected.
+ *
+ * @param {WPDataRegistry} registry Data registry.
+ * @return {WPDataRegistry} Modified data registry.
+ */
+function toggleBlockSelectedStyles( registry ) {
+	return {
+		dispatch: ( namespace ) => {
+			const namespaceName =
+				typeof namespace === 'string' ? namespace : namespace.name;
+			const actions = { ...registry.dispatch( namespaceName ) };
+
+			const originalSelectBlockAction = actions.selectBlock;
+			actions.selectBlock = ( ...args ) => {
+				_toggleBlockSelectedClass( true );
+				return originalSelectBlockAction( ...args );
+			};
+
+			const originalClearSelectedBlockAction = actions.clearSelectedBlock;
+			actions.clearSelectedBlock = ( ...args ) => {
+				_toggleBlockSelectedClass( false );
+				return originalClearSelectedBlockAction( ...args );
+			};
+
+			return actions;
+		},
+	};
+}
+
+window.wp.data.use( toggleBlockSelectedStyles );
+
+// The editor-canvas iframe relies upon `srcdoc`, which does not trigger a
+// `load` event. Thus, we must poll for the iframe to be ready.
+let overrideAttempts = 0;
+const overrideInterval = setInterval( () => {
```

Comment:
I explored a few different approaches for awaiting the `iframe` render. Unfortunately, none of them succeeded.

- `iframe.onload` does not work when the `iframe` uses `srcdoc` instead of `src`. 
- Awaiting a "ready Gutenberg" via [subscribing to the store](https://github.com/WordPress/gutenberg/blob/a7eb9a220f7105d6a194b0e0228fbbe2484ae727/packages/react-native-bridge/common/gutenberg-web-single-block/content-functions.js#L60-L68) runs before the `iframe` renders.